tidy3d/components/medium.py

```diff
@@ -93,6 +137,12 @@ class AbstractMedium(ABC, Tidy3dBaseModel):
         "useful in some cases.",
     )
 
+    nonlinear_spec: None = pd.Field(
```

Comment:
just wanted to clarify how this works: would the user just get an error if they try to plug `Chi3Spec` into a dispersive medium? Ideally we should have some meaningful error message something like `f"Nonlinear feature is not currently supported for medium class {type(cls)}"`, but it's up to you

Btw, `CustomIsotropicMedium` (which will become just `CustomMedium` in 3.0) inherits from `Medium`. Do we support `Chi3Spec` for custom mediums on backend?